teps/0103-skipping-reason.md

```diff
@@ -0,0 +1,215 @@
+---
+status: implementable
+title: Skipping Reason
+creation-date: '2022-04-06'
+last-updated: '2022-04-06'
+authors:
+- '@jerop'
+see-also:
+- TEP-0007
+- TEP-0059
+- TEP-0100
+---
+
+# TEP-0103: Skipping Reason
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Example](#example)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+Today, users only know that a `PipelineTask` was skipped, but they don't know for which
+exact reason. This can be confusing when debugging `Pipelines`. In this TEP, we propose
+adding the reason for skipping to the `SkippedTasks` field in `PipelineRunStatus` to
+improve usability and debuggability.
+
+## Motivation
+
+There are many [reasons][reasons] why a `PipelineTask` could be skipped, including:
+* at least one of its `when` expressions evaluated to false 
+* at least one of its `Conditions` failed
```

Comment:
As `Conditions` are deprecated and will soon be removed, should we even try to support that ? 😛